app/imports/components/posts/NewPost.jsx

```diff
@@ -8,78 +9,86 @@ import { distanceFrom } from '/imports/lib/geometry.js';
 import sanitizeHtml from 'sanitize-html';
 
 import { addPost } from '/imports/api/photosMethods.js';
-
 import { LocationNotice } from '/imports/components/posts/LocationNotice.jsx';
 
 export class NewPost extends Component {
   constructor(props) {
     super(props);
 
     let fingerprint = Session.get('fingerprint');
-    let photo = Session.get('newPhoto');
+    let originalImageSrc = Session.get('newPhoto');
 
     this.state = {
-      processingPhoto: true,
-      originalPhoto: photo.image,
-      exif: photo.exif,
       fingerprint,
-      locationApproved: false,
+      processingPhoto: true,
       locationChecking: true,
-      locationUnavailable: false,
-      imageReady: false,
+      locationApproved: false,
       uploading: false
     };
 
+    this.originalImageSrc = originalImageSrc;
+
     this.checkGeofence = this.checkGeofence.bind(this);
 
     this.onInputChange = this.onInputChange.bind(this);
     this.onSubmitForm = this.onSubmitForm.bind(this);
     this.handleUpload = this.handleUpload.bind(this);
-    this.locationUnavailable = this.locationUnavailable.bind(this);
-    this.startProcessing = this.startProcessing.bind(this);
-    this.processPhoto = this.processPhoto.bind(this);
+    this.getOrientation = this.getOrientation.bind(this);
+    this.handleExif = this.handleExif.bind(this);
+    this.processImage = this.processImage.bind(this);
     this.resizeImage = this.resizeImage.bind(this);
     this.imageResizedCallback = this.imageResizedCallback.bind(this);
 
   }
 
   componentWillMount() {
-    navigator.geolocation.getCurrentPosition(this.checkGeofence, this.locationUnavailable, {timeout:10000});
-    this.startProcessing();
-  }
-
-  locationUnavailable() {
-    this.setState({
-      'locationChecking': false,
-      'locationUnavailable': true,
-    });
+    navigator.geolocation.getCurrentPosition(this.checkGeofence);
+    this.rebuildFile();
   }
 
-  startProcessing() {
-    let photo = Session.get('newPhoto');
+  rebuildFile() {
+    let imageSrc = this.originalImageSrc;
 
-    if(photo.image === undefined) {
+    if(imageSrc === undefined) {
       FlowRouter.go('home');
     }
 
     this.phantomImage = new Image();
-    this.phantomImage.onload = this.processPhoto;
-    this.phantomImage.src =  photo.image;
+    this.phantomImage.onload = this.processImage;
+    this.phantomImage.src =  imageSrc;
+  }
+
+  processImage() {
+    if(!this.imageOrientation) {
+      this.getOrientation();
+    } else {
+      this.resizeImage();
+    }
   }
 
-  processPhoto() {
-    this.resetOrientation();
-    this.resizeImage();
+  getOrientation() {
+    EXIF.getData(this.phantomImage, this.handleExif);
+  }
+
+  handleExif() {
+    let orientation = EXIF.getTag(this.phantomImage, "Orientation");
+
+    if(orientation) {
+      this.imageOrientation = orientation;
+      this.resetOrientation();
```

Comment:
this function should be named more precisely: `rotateImage` or `fixImageOrientation` seems a better to me